ipn/ipnlocal/local.go

```diff
@@ -2358,7 +2360,14 @@ func (b *LocalBackend) enterState(newState ipn.State) {
 		if authURL == "" {
 			systemd.Status("Stopped; run 'tailscale up' to log in")
 		}
-	case ipn.Starting, ipn.NeedsMachineAuth:
+	case ipn.Starting:
+		if oldState == ipn.NeedsLogin && !keyExpired && blocked {
+			// This means that key was extended **after** expiry,
+			// unblock engine updates.
+			b.blockEngineUpdates(false)
+		}
+		fallthrough
```

Comment:
this isn't intuitive on its own (I understand how you got here from the old code, but people won't have that context later). e.g. How's Starting related to NeedsMachineAuth?

I'd rather you just copy/paste those two lines if we need them in this case.

Reply:
makes sense, copied the two lines.